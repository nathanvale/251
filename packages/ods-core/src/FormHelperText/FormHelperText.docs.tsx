/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { FormHelperText, FormHelperTextProps } from "./FormHelperText";

export const docs: ComponentDocs<FormHelperTextProps> = {
  category: "Content",
  componentName: "FormHelperText",
  description:
    "This is a low level and atomic component used to show descriptive helper text under form components.",
  propDescriptions: {
    id: "The unique id of the component, required to accessibility",
  },
  migrationGuide: false,
  examples: [
    {
      label: "Simple use case",
      description: [
        "Simple use case of a FormHelperText. ",
        "The id is mandatory as it has to be pointed to by the corresponding input component",
      ].join(" "),
      Code: () => (
        <FormHelperText id="controlHelperText">
          This is a helper text
        </FormHelperText>
      ),
    },
    {
      label: "data-id",
      description:
        "If data-id is provided, it will be set on the component, otherwise it will be set to 'helperText'.",
      Code: () => (
        <FormHelperText id="controlHelperText">
          This is a helper text
        </FormHelperText>
      ),
    },
  ],
  snippets: [
    {
      label: "Basic",
      Code: () => (
        <FormHelperText id="controlHelperText" data-id="controlHelperText">
          This is a helper text
        </FormHelperText>
      ),
    },
  ],
};
