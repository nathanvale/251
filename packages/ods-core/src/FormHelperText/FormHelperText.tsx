import * as React from "react";
import MuiFormHelperText, {
  FormHelperTextProps as MuiFormHelperTextProps,
} from "@material-ui/core/FormHelperText";
import { BaseFormStateProps } from "@origin-digital/ods-types";

export interface FormHelperTextProps extends BaseFormStateProps {
  id: string;
  muiProps?: MuiFormHelperTextProps;
}

export const FormHelperText = ({
  children,
  muiProps,
  ...others
}: FormHelperTextProps) => (
  <MuiFormHelperText {...muiProps} {...others}>
    {children}
  </MuiFormHelperText>
);

FormHelperText.displayName = "FormHelperText";
FormHelperText.defaultProps = {
  "data-id": "formHelperText",
};
