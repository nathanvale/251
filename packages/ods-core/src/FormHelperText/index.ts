export { FormHelperText } from "./FormHelperText";
export type FormHelperTextProps = import("./FormHelperText").FormHelperTextProps;
