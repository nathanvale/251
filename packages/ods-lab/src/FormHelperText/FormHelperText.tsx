import * as React from "react";
import MuiFormHelperText, {
  FormHelperTextProps as IFormHelperText,
} from "@material-ui/core/FormHelperText";

export interface FormHelperTextProps extends IFormHelperText {
  "data-id"?: string;
}

export const FormHelperText = (props: FormHelperTextProps) => (
  <MuiFormHelperText {...props}>{props.children}</MuiFormHelperText>
);
