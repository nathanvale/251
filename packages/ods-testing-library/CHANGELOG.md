# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
