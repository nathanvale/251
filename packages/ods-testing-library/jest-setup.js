import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
