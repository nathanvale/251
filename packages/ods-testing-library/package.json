{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
