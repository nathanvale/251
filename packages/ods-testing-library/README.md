# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
