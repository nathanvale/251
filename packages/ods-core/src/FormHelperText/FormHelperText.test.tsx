import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { FormHelperText } from "./FormHelperText";

test("it renders the text as a paragraph", () => {
  const { container } = render(
    <FormHelperText id="helperText" data-id="helperTextDataId">
      This is a simple helper text
    </FormHelperText>,
  );

  expect(container).toMatchSnapshot();
});

test("it passes down id, data-id and children text", () => {
  const { getAllByText } = render(
    <FormHelperText id="helperText" data-id="helperTextDataId">
      This is a simple helper text
    </FormHelperText>,
  );

  const items = getAllByText("This is a simple helper text");

  expect(items).toHaveLength(1);
  expect(items[0].getAttribute("id")).toEqual("helperText");
  expect(items[0].getAttribute("data-id")).toEqual("helperTextDataId");
});
