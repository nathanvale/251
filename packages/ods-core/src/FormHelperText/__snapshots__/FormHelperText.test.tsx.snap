// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the text as a paragraph 1`] = `
<div>
  <p
    class="MuiFormHelperText-root"
    data-id="helperTextDataId"
    id="helperText"
  >
    This is a simple helper text
  </p>
</div>
`;
